tools/topology/topology2/cavs-hdmi-in-ssp.conf

```diff
@@ -0,0 +1,76 @@
+<searchdir:include>
+<searchdir:include/common>
+<searchdir:include/components>
+<searchdir:include/dais>
+<searchdir:include/pipelines/cavs>
+<searchdir:platform>
+<searchdir:platform/intel>
+
+<vendor-token.conf>
+<manifest.conf>
+<pdm_config.conf>
+<tokens.conf>
+<virtual.conf>
+<host-gateway-playback.conf>
+<host-gateway-capture.conf>
+<io-gateway.conf>
+<io-gateway-capture.conf>
+<host-copier-gain-mixin-playback.conf>
+<mixout-gain-dai-copier-playback.conf>
+<deepbuffer-playback.conf>
+<dai-copier-be.conf>
+<dai-copier-eqiir-module-copier-capture.conf>
+<gain-capture.conf>
+<gain-module-copier.conf>
+<data.conf>
+<pcm.conf>
+<pcm_caps.conf>
+<fe_dai.conf>
+<ssp.conf>
+<dmic.conf>
+<hda.conf>
+<intel/hw_config_cardinal_clk.conf>
+<manifest.conf>
+<route.conf>
+<intel/common_definitions.conf>
+<dai-copier.conf>
+<module-copier.conf>
+<pipeline.conf>
+<dai.conf>
+<host.conf>
+<dmic-default.conf>
+<hdmi-default.conf>
+<hdmi-in-default.conf>
+<input_pin_binding.conf>
+<output_pin_binding.conf>
+<input_audio_format.conf>
+<output_audio_format.conf>
+
+Define {
+	# override HDMI-IN/DMIC default definitions
+	HDMI_IN_1_ID		0
+	HDMI_IN_2_ID		1
+	DMIC0_ID			3
+	DMIC1_ID			4
+	# override HDMI BE link ids
+	NUM_HDMIS			3
+	HDMI1_ID			5
+	HDMI2_ID			6
+	HDMI3_ID			7
+	PLATFORM 			"none"
+}
+
+# override defaults with platform-specific config
+IncludeByKey.PLATFORM {
+	"mtl"	"platform/intel/mtl.conf"
+}
+
+# include HDMI-in capture config.
+IncludeByKey.HDMI_IN_CAPTURE {
```

Comment:
do we really need a new file for this? Can we not use sof-hda-generic and just add this IncludeByKey block alone? and set the BE link ID's in the tplg_targets.cmake file to generate the new tplg?

Reply:
Hi @ranj063,
 
No plan to add this No codec + HDMI-in capture support with HD audio. same time don't want to mix with any other I2S machine driver(ex: Es8326/RT5682). Since this no SSP codec+HDMI-In capture+HDMI audio playback is aligned with sof_ssp_amp machine driver and reuses the same for MTL. Also the plan is going forward for topology2 supported platforms will use this file to support no SSP codec+ HDMI-in capture via I2S + HDMI audio playback by using sof_ssp_amp.c machine driver.
Basically the intention is not to disturb/reduce complexity on I2S codec machine drivers and use the sof_ssp_amp.c machine driver for this use case.